com.ibm.wala.cast.java.ecj/src/com/ibm/wala/cast/java/translator/jdt/JDTJava2CAstTranslator.java

```diff
@@ -572,8 +572,8 @@ else if (met.getParameterTypes().length > 0)
       // this.
 
       List<CAstNode> bodyNodes = new ArrayList<>(staticInits.size());
-      for (int i = 0; i < staticInits.size(); i++)
-        bodyNodes.add(visitFieldInitNode(staticInits.get(i), newContext));
+      for (ASTNode staticInit : staticInits)
```

Comment:
Do we know what kind of bytecodes this generates?  I think the modified code will allocate and use an iterator, rather than using `get()` calls.  In performance-critical code this could cause a slight performance degradation.  But I'm not 100% sure.  For arrays I think the same code comes out so we should be fine.

Reply:
Consensus opinion I am seeing online (e.g., [here](https://stackoverflow.com/q/9226483/91929) or [here](https://stackoverflow.com/q/256859/91929) is that the two variants are equally efficient, or that the for-each variant might be slightly faster by computing the upper bound just once.

Reply:
Ok.  I'm not 100% convinced, but we should have real benchmarks if we are worried about this stuff.  Probably in most perf-critical spots we are already using arrays anyway.